crates/bevy_render/src/render_ecs_resource.rs

```diff
@@ -0,0 +1,54 @@
+use std::marker::PhantomData;
+
+use bevy_app::{App, Plugin};
+use bevy_ecs::system::{
+    lifetimeless::{SCommands, SRes},
+    Resource, RunSystem, SystemParamItem,
+};
+pub use bevy_render_macros::ExtractResource;
+
+use crate::{RenderApp, RenderStage};
+
+/// Describes how a resource gets extracted for rendering.
+///
+/// Therefore the resource is transferred from the "main world" into the "render world"
+/// in the [`RenderStage::Extract`](crate::RenderStage::Extract) step.
+pub trait ExtractResource: Resource {
+    /// Defines how the resource is transferred into the "render world".
+    fn extract_resource(res: &Self) -> Self;
+}
+
+/// This plugin extracts the resources into the "render world".
+///
+/// Therefore it sets up the [`RenderStage::Extract`](crate::RenderStage::Extract) step
+/// for the specified [`Resource`].
+pub struct ExtractResourcePlugin<R: ExtractResource + Resource>(PhantomData<R>);
+
+impl<R: ExtractResource + Resource> Default for ExtractResourcePlugin<R> {
+    fn default() -> Self {
+        Self(PhantomData)
+    }
+}
+
+impl<R: ExtractResource + Resource> Plugin for ExtractResourcePlugin<R> {
+    fn build(&self, app: &mut App) {
+        let system = ExtractResourceSystem::<R>::system(&mut app.world);
+        if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
+            render_app.add_system_to_stage(RenderStage::Extract, system);
+        }
+    }
+}
+
+/// This system extracts the resource of the corresponding [`Resource`] type
+/// by cloning it.
+pub struct ExtractResourceSystem<R: ExtractResource + Resource>(PhantomData<R>);
+
+impl<R: ExtractResource + Resource> RunSystem for ExtractResourceSystem<R> {
```

Comment:
`Resource` is already a supertrait of `ExtractResource`, which makes it unnecessary to mention it again here.
```suggestion
pub struct ExtractResourcePlugin<R: ExtractResource>(PhantomData<R>);

impl<R: ExtractResource> Default for ExtractResourcePlugin<R> {
    fn default() -> Self {
        Self(PhantomData)
    }
}

impl<R: ExtractResource> Plugin for ExtractResourcePlugin<R> {
    fn build(&self, app: &mut App) {
        let system = ExtractResourceSystem::<R>::system(&mut app.world);
        if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
            render_app.add_system_to_stage(RenderStage::Extract, system);
        }
    }
}

/// This system extracts the resource of the corresponding [`Resource`] type
/// by cloning it.
pub struct ExtractResourceSystem<R: ExtractResource>(PhantomData<R>);

impl<R: ExtractResource> RunSystem for ExtractResourceSystem<R> {
```